src/main/java/seedu/address/logic/recommender/Recommender.java

```diff
@@ -0,0 +1,134 @@
+
+        List<List<Location>> locationRecommendations = timingRecommendations.stream()
+                .map(this::getLocationsFromHourBlock)
+                .map(lr::recommend)
+                .collect(Collectors.toList());
+
+        List<Recommendation> recommendations = CollectionUtil
+                .zip(locationRecommendations.stream(),
+                        timingRecommendations.stream(),
+                        this::recommendFromLocationsHourBlock)
+                .flatMap(List::stream)
+                .collect(Collectors.toList());
+
+        List<Recommendation> filteredRecommendations = filterRecommendations(recommendations);
+
+        return filteredRecommendations.stream()
+                .sorted().limit(RECOMMENDATION_LIMIT)
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * Sets up the {@code LocationRecommender}, {@code TimingRecommender}
+     * and {@code LocationTracker} for each person.
+     */
+    private void initialise(Collection<ContactIndex> contactIndices, Collection<Location> destinations) {
+        lr.initialise(destinations);
+        sc.initialise(contactIndices);
+        locationTrackers = sc.getParticipants().stream()
+                .map(LocationTracker::new)
+                .collect(Collectors.toSet());
+    }
+
+    /**
+     * Gets location from hour block.
+     */
+    private Set<Location> getLocationsFromHourBlock(HourBlock hourBlock) {
+        return locationTrackers.stream()
+                .map(lt -> lt.getLocation(hourBlock))
+                .filter(Optional::isPresent)
+                .map(Optional::get)
+                .collect(Collectors.toSet());
+    }
+
+    /**
+     * Returns a list of recommendation based on a particular timing.
+     */
+    private List<Recommendation> recommendFromLocationsHourBlock(List<Location> locations, HourBlock hourBlock) {
+        return locations.stream()
+                .map(l -> new Recommendation(l, hourBlock))
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * Filters out duplicate timings or locations.
+     */
+    private List<Recommendation> filterRecommendations(List<Recommendation> recommendations) {
+        Set<TimePeriod> timePeriods = new HashSet<>();
+        Set<Location> locations = new HashSet<>();
+
+        List<Recommendation> filteredRecommendations = new ArrayList<>();
+
+        for (Recommendation recommendation : recommendations) {
+            TimePeriod timePeriod = recommendation.getTimePeriod();
+            Location location = recommendation.getLocation();
+
+            if (timePeriods.contains(timePeriod) || locations.contains(location)) {
+                continue;
+            }
+
+            filteredRecommendations.add(recommendation);
+            timePeriods.add(timePeriod);
+            locations.add(location);
+        }
+
+        return filteredRecommendations;
+    }
```

Comment:
With the way it was set up, the recommendations were just the same general location at the same time. In this way, we can introduce more varied recommendations.